builtin/game/falling.lua

```diff
@@ -56,78 +48,91 @@ core.register_entity(":__builtin:falling_node", {
 			self.object:set_acceleration({x = 0, y = -10, z = 0})
 		end
-		-- Turn to actual node when colliding with ground, or continue to move
+
 		local pos = self.object:get_pos()
 		-- Position of bottom center point
-		local bcp = {x = pos.x, y = pos.y - 0.7, z = pos.z}
-		-- 'bcn' is nil for unloaded nodes
-		local bcn = core.get_node_or_nil(bcp)
+		local below_pos = {x = pos.x, y = pos.y - 0.7, z = pos.z}
+		-- Avoid bugs caused by an unloaded node below
+		local below_node = core.get_node(below_pos)
 		-- Delete on contact with ignore at world edges
-		if bcn and bcn.name == "ignore" then
+		if below_node.name == "ignore" then
 			self.object:remove()
 			return
 		end
-		local bcd = bcn and core.registered_nodes[bcn.name]
-		if bcn and
-				(not bcd or bcd.walkable or
-				(core.get_item_group(self.node.name, "float") ~= 0 and
-				bcd.liquidtype ~= "none")) then
-			if bcd and bcd.leveled and
-					bcn.name == self.node.name then
-				local addlevel = self.node.level
-				if not addlevel or addlevel <= 0 then
-					addlevel = bcd.leveled
-				end
-				if core.add_node_level(bcp, addlevel) == 0 then
-					self.object:remove()
-					return
-				end
-			elseif bcd and bcd.buildable_to and
-					(core.get_item_group(self.node.name, "float") == 0 or
-					bcd.liquidtype == "none") then
-				core.remove_node(bcp)
+
+		local below_nodef = core.registered_nodes[below_node.name]
+		-- Is it a level node we can add to?
+		if below_nodef
+		and below_nodef.leveled
+		and below_node.name == self.node.name then
+			local addlevel = self.node.level
+			if not addlevel or addlevel <= 0 then
+				addlevel = below_nodef.leveled
+			end
+			if core.add_node_level(below_pos, addlevel) == 0 then
+				self.object:remove()
 				return
 			end
-			local np = {x = bcp.x, y = bcp.y + 1, z = bcp.z}
-			-- Check what's here
-			local n2 = core.get_node(np)
-			local nd = core.registered_nodes[n2.name]
-			-- If it's not air or liquid, remove node and replace it with
-			-- it's drops
-			if n2.name ~= "air" and (not nd or nd.liquidtype == "none") then
-				core.remove_node(np)
-				if nd and nd.buildable_to == false then
-					-- Add dropped items
-					local drops = core.get_node_drops(n2, "")
+		end
+
+		-- Stop node if it falls on walkable surface, or floats on water
+		if (below_nodef and below_nodef.walkable == true)
+		or (below_nodef
+				and core.get_item_group(self.node.name, "float") ~= 0
+				and below_nodef.liquidtype ~= "none") then
+
+			self.object:set_velocity({x = 0, y = 0, z = 0})
+		end
+
+		-- Has the fallen node stopped moving ?
+		local vel = self.object:get_velocity()
+		if vector.equals(vel, {x = 0, y = 0, z = 0}) then
+			local npos = self.object:get_pos()
+			-- Get node we've landed inside
+			local cnode = minetest.get_node(npos).name
+			local cdef = core.registered_nodes[cnode]
+			-- If 'air' or buildable_to or an attached_node then place node,
+			-- otherwise drop falling node as an item instead.
+			if cnode == "air"
```

Comment:
Here check for 'air_equivalent = true' instead ("air" has this node parameter set to true also).

Reply:
https://github.com/minetest/minetest/blob/0e306c084284aeafb3d5cde0cfec11a85a11cb9c/builtin/game/register.lua#L344

Reply:
Done.